pkg/backend/httpstate/state.go

```diff
@@ -48,48 +48,51 @@ type tokenSource struct {
 	done     chan bool
 }
 
-func newTokenSource(ctx context.Context, token string, backend *cloudBackend, update client.UpdateIdentifier,
+func newTokenSource(ctx context.Context, initialToken string, backend *cloudBackend, update client.UpdateIdentifier,
 	duration time.Duration) (*tokenSource, error) {
 
-	// Perform an initial lease renewal.
-	newToken, err := backend.client.RenewUpdateLease(ctx, update, token, duration)
-	if err != nil {
-		return nil, err
-	}
-
 	requests, done := make(chan tokenRequest), make(chan bool)
 	go func() {
 		// We will renew the lease after 50% of the duration has elapsed to allow more time for retries.
 		ticker := time.NewTicker(duration / 2)
 		defer ticker.Stop()
 
+		token := initialToken
+		var currentError error
+
+		renew := func() {
+			newToken, err := backend.client.RenewUpdateLease(ctx, update, token, duration)
+
+			// If we get an error from the backend, leave `err` set and surface it during
+			// the next request for a lease token.
+			if err != nil {
+				logging.V(3).Infof("error renewing lease: %v", err)
+				currentError = fmt.Errorf("renewing lease: %w", err)
+
+				ticker.Stop()
+			} else {
+				token = newToken
+			}
+		}
+
+		// Is an initial lease renewal?
+		// renew()
```

Comment:
This one is tricky? The original code was doing `RenewUpdateLease` call but ignoring the resulting `newToken` completely. It seems that the `token string` this func receives as a parameter is already a valid token coming out of the API. So maybe we should not do the initial lease renewal at all?

Reply:
Ah, it's possible that the call was to extend the lifetime of the initial token.
Say `initialToken` is guaranteed valid, but it expires soon. The initial `renew` call might be to extend its expiry? That might be a question for the Service team?

Reply:
As far as I understand from the trace and digging in the code, the token was *just acquired* couple hundred ms ago from another service call. So it seems like we're ok not renewing right away? If we need to though, I can uncomment the lines here and it will start by renewing, but still async (goroutine). This preserves the speedup we want because this call becomes concurrent with another service call.